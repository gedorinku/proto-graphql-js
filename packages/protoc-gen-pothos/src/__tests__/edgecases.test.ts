import { itGeneratesDSLsToMatchSnapshtos } from "./__helpers__/process.test.helper";

describe("import squashed union", () => {
  itGeneratesDSLsToMatchSnapshtos("edgecases/import_squashed_union/pkg1", [
    "edgecases/import_squashed_union/pkg1/types_pb_nexus.ts",
  ]);
  itGeneratesDSLsToMatchSnapshtos("edgecases/import_squashed_union/pkg2", [
    "edgecases/import_squashed_union/pkg2/types_pb_nexus.ts",
  ]);
});
