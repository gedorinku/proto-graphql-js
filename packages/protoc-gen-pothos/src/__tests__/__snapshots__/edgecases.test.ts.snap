// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import squashed union with ts-proto generates pothos DSLs: edgecases/import_squashed_union/pkg1/types.pb.pothos.ts 1`] = `
"// Code generated by protoc-gen-pothos. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg1/types.proto

/* eslint-disable */

import { builder } from \\"../../../../builder\\";
import * as _$testapis$edgecases$import_squashed_union$pkg1$types from \\"./testapis/edgecases/import_squashed_union/pkg1/types\\";
export const OneofMessage1$Ref = builder.objectRef<_$testapis$edgecases$import_squashed_union$pkg1$types.OneofMessage1>(\\"OneofMessage1\\");
builder.objectType(OneofMessage1$Ref, {
    name: \\"OneofMessage1\\",
    fields: t => ({
        body: t.expose(\\"body\\", {
            type: \\"String\\",
            nullable: false
        })
    }),
    isTypeOf: source => {
        return (source as _$testapis$edgecases$import_squashed_union$pkg1$types.OneofMessage1 | {
            $type: string & {};
        }).$type === \\"testapis.edgecases.import_squashed_union.pkg1.OneofMessage1\\";
    }
});
export type OneofMessage1Input$Shape = {
    body: _$testapis$edgecases$import_squashed_union$pkg1$types.OneofMessage1[\\"body\\"];
};
export const OneofMessage1Input$Ref = builder.inputRef<OneofMessage1Input$Shape>(\\"OneofMessage1Input\\");
OneofMessage1Input$Ref.implement({
    fields: t => ({
        body: t.field({
            type: \\"String\\",
            required: true
        })
    })
});
export type SquashedOneofInput$Shape = {
    msg1?: OneofMessage1Input$Shape | null;
};
export const SquashedOneofInput$Ref = builder.inputRef<SquashedOneofInput$Shape>(\\"SquashedOneofInput\\");
SquashedOneofInput$Ref.implement({
    fields: t => ({
        msg1: t.field({
            type: OneofMessage1Input$Ref,
            required: false
        })
    })
});
export const SquashedOneof$Ref = builder.unionType(\\"SquashedOneof\\", {
    types: [
        OneofMessage1$Ref
    ]
});
"
`;

exports[`import squashed union with ts-proto generates pothos DSLs: edgecases/import_squashed_union/pkg2/types.pb.pothos.ts 1`] = `
"// Code generated by protoc-gen-pothos. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg2/types.proto

/* eslint-disable */

import { builder } from \\"../../../../builder\\";
import * as __$pkg1$types_pb_pothos from \\"../pkg1/types.pb.pothos\\";
import { SquashedOneof$Ref } from \\"../pkg1/types.pb.pothos\\";
import { SquashedOneofInput$Ref } from \\"../pkg1/types.pb.pothos\\";
import { SquashedOneofInput$Shape } from \\"../pkg1/types.pb.pothos\\";
import * as _$testapis$edgecases$import_squashed_union$pkg1$types from \\"./testapis/edgecases/import_squashed_union/pkg1/types\\";
import * as _$testapis$edgecases$import_squashed_union$pkg2$types from \\"./testapis/edgecases/import_squashed_union/pkg2/types\\";
export const Message$Ref = builder.objectRef<_$testapis$edgecases$import_squashed_union$pkg2$types.Message>(\\"Message\\");
builder.objectType(Message$Ref, {
    name: \\"Message\\",
    fields: t => ({
        msg: t.field({
            type: SquashedOneof$Ref,
            nullable: true,
            resolve: source => {
                const value = source.msg?.msg1;
                return value;
            }
        })
    }),
    isTypeOf: source => {
        return (source as _$testapis$edgecases$import_squashed_union$pkg2$types.Message | {
            $type: string & {};
        }).$type === \\"testapis.edgecases.import_squashed_union.pkg2.Message\\";
    }
});
export type MessageInput$Shape = {
    msg?: SquashedOneofInput$Shape | null;
};
export const MessageInput$Ref = builder.inputRef<MessageInput$Shape>(\\"MessageInput\\");
MessageInput$Ref.implement({
    fields: t => ({
        msg: t.field({
            type: SquashedOneofInput$Ref,
            required: false
        })
    })
});
"
`;
